pkg/kubectl/metricsutil/metrics_client.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metricsutil
+
+import (
+	"encoding/json"
+	"fmt"
+
+	metrics_api "k8s.io/heapster/metrics/apis/metrics/v1alpha1"
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	MetricsRoot              = "/apis/metrics/v1alpha1/"
+	DefaultHeapsterNamespace = "kube-system"
+	DefaultHeapsterScheme    = "http"
+	DefaultHeapsterService   = "heapster"
+	DefaultHeapsterPort      = "" // use the first exposed port on the service
+)
+
+type HeapsterMetricsClient struct {
+	Client            *client.Client
+	HeapsterNamespace string
+	HeapsterScheme    string
+	HeapsterService   string
+	HeapsterPort      string
+}
+
+func NewHeapsterMetricsClient(client *client.Client, namespace, scheme, service, port string) *HeapsterMetricsClient {
+	return &HeapsterMetricsClient{
+		Client:            client,
+		HeapsterNamespace: namespace,
+		HeapsterScheme:    scheme,
+		HeapsterService:   service,
+		HeapsterPort:      port,
+	}
+}
+
+func DefaultHeapsterMetricsClient(client *client.Client) *HeapsterMetricsClient {
+	return NewHeapsterMetricsClient(client, DefaultHeapsterNamespace, DefaultHeapsterScheme, DefaultHeapsterService, DefaultHeapsterPort)
+}
+
+func PodMetricsUrl(namespace string, name string) string {
+	return fmt.Sprintf("%s/namespaces/%s/pods/%s", MetricsRoot, namespace, name)
```

Comment:
You need to validate that this URL is well formed.  Same with the Node ones below.